evalai/submissions.py

```diff
@@ -140,3 +154,45 @@ def push(image, phase):
                     if k != "progressDetail"
                 )
             )
+
+
+@click.command()
+@click.argument("URL", nargs=1)
+def download_file(url):
+    parsed_url = urlparse.urlparse(url)
+    bucket = urlparse.parse_qs(parsed_url.query).get("bucket")
+    key = urlparse.parse_qs(parsed_url.query).get("key")
+    request_path = URLS.download_file.value
+    request_path = request_path.format(bucket[0], key[0])
+    response = make_request(request_path, "GET")
+    signed_url = response.get("signed_url")
+    file_name = key[0].split("/")[-1]
```

Comment:
This will fail if the URL is not of the format that is specified. Can we do bunch of initial checks:

1. Hostname should be the CLI's host name
2. It should contain bucket and key as two query params
3. Something else...

Reply:
We can't have the hostname check for now since the hostname in download_url and evalai backend are differnt.

Reply:
Don't you think that there should be check for valid URL? If someone enters a wrong URL, then CLI will not handle it.

Moreover, I am not sure if I understand what you mean by the above sentence. Can you please give an example?

Reply:
I agree that there should be a check for a hostname.
Suppose, the user will enter a URL - 
`https://evalai.cloudcv.org/web/submission-files/?bucket=<blah>&key=<blah>`
but the hostname set in the `cli` will be `https://evalapi.cloudcv.org` which doesn't match the entered URL's hostname.
I didn't want to hardcode the URL for checking but I think we should it do it now and can remove that later.

Reply:
Thanks for the clarification. Looking forward to the updated code.